app/tasks/mailchimp.py

```diff
@@ -38,6 +43,174 @@ def get_client():
+            .format(list_id, e, e.response))
+        rollbar.report_exc_info()
+        raise e
+
+
+def send_document_expiry_campaign(client, sellers):
+    folder_id = getenv('MAILCHIMP_MARKETPLACE_FOLDER_ID')
+    if not folder_id:
+        raise MailChimpConfigException('Failed to get MAILCHIMP_MARKETPLACE_FOLDER_ID from the environment variables.')
+
+    list_id = getenv('MAILCHIMP_SELLER_LIST_ID')
+    if not list_id:
+        raise MailChimpConfigException('Failed to get MAILCHIMP_SELLER_LIST_ID from the environment variables.')
+
+    title = 'Expiring documents - {}'.format(pendulum.today().to_date_string())
+
+    sent_expiring_documents_audit_event = audit_service.filter(
+        AuditEvent.type == audit_types.sent_expiring_documents_email.value,
+        AuditEvent.data['campaign_title'].astext == title
+    ).one_or_none()
+
+    if (sent_expiring_documents_audit_event > 0):
+        return
+
+    conditions = []
+    for seller in sellers:
+        for email_address in seller['email_addresses']:
+            conditions.append({
+                'condition_type': 'EmailAddress',
+                'op': 'is',
+                'field': 'EMAIL',
+                'value': email_address
+            })
+
+    recipients = {
+        'list_id': list_id,
+        'segment_opts': {
+            'match': 'any',
+            'conditions': conditions
+        }
+    }
+
+    settings = {
+        'folder_id': folder_id,
+        'preview_text': 'Please update your documents',
+        'subject_line': 'Your documents are soon to expire or have expired',
+        'title': title
+    }
+
+    campaign = create_campaign(client, recipients, settings)
+
+    template = template_env.get_template('mailchimp_document_expiry.html')
+    email_body = template.render(current_year=pendulum.today().year)
+    update_campaign_content(client, campaign['id'], email_body)
+
+    send_campaign(client, campaign['id'])
+
+    audit_service.log_audit_event(
+        audit_type=audit_types.sent_expiring_documents_email,
+        data={
+            'campaign_title': title,
+            'sellers': sellers
+        },
+        db_object=None,
+        user=None
+    )
+
+
+@celery.task
+def send_document_expiry_reminder():
+    # Find sellers with documents 28 days from expiry, 14 days from expiry, on expiry and 28 days after expiry
+    sellers = (suppliers.get_suppliers_with_expiring_documents(days=28) +
+               suppliers.get_suppliers_with_expiring_documents(days=14) +
+               suppliers.get_suppliers_with_expiring_documents(days=0) +
+               suppliers.get_suppliers_with_expiring_documents(days=-28))
+
+    if not sellers:
+        return
+
+    try:
+        client = get_client()
+        send_document_expiry_campaign(client, sellers)
+    except Exception as e:
+        current_app.logger.error(
+            'An error occurred while sending the expiring documents campaign, aborting: {} {}'
+            .format(e, e.response))
```

Comment:
Sorry, missed this in chat - I think we could have issues with `e.response` here as given it is catching all exceptions and not just a particular type of exception, not all may provide this value. Instead I think `format(str(e))` would do what you want, also removing one `{}` in the string being formatted.

Reply:
I've removed the reference to `response`.  After some testing it looks like `.format(e)` is sufficient.  This format is used elsewhere in the repo too.